### PR TITLE
Benchmark for cellfinder workflow with CLI arguments

### PR DESCRIPTION
**Pull request by sfmig**
Adds `asv` benchmarks to time the cellfinder workflows defined in PR #23.

We benchmark the full workflow and its individual steps (reading, detecting cells and saving results to file). 

This PR also includes some refactoring changes to the cellfinder workflows, to allow us to split the setup steps into data downloading and other steps. This way we can download the data only once for all the repeats of a benchmark.

Would need rebasing after PR #23 is squash-and-merged

To run the benchmarks, run `asv run` from the directory where `asv.conf.json`

### CODE REVIEW

File: asv.conf.json [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    // ],
+
+    // Customizable commands for installing and uninstalling the project.
+    // See asv.conf.json documentation.
+    "install_command": ["in-dir={env_dir} python -mpip install --force-reinstall {wheel_file}"],
+    "uninstall_command": ["return-code=any python -mpip uninstall -y {project}"],
+
+    // List of branches to benchmark. If not provided, defaults to "master"
+    // (for git) or "default" (for mercurial).
+    "branches": ["smg/cellfinder-workflow"], // for git
```
Comment (alessandrofelder) [inline asv.conf.json:42]: Is this right? (It's a different name to the branch of the PR and I had to change it to run it locally?)

**Review by alessandrofelder — APPROVED** [review]
Tiny comment that I am not sure about.
Otherwise LGTM :rocket:

**Comment by alessandrofelder** [comment]
(Also, remember to rebase I guess @sfmig ?)

File: asv.conf.json (hunk 1 above)
Comment (sfmig) [reply asv.conf.json:42]: yes sorry, I think that came about from the rebasing and merging of conflicts 🤔

**Pull request merged**